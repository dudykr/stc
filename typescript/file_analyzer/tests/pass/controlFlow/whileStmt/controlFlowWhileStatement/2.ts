export declare let x: string | number;
x = "";
while (x = x.length) {
    x; // number
    x = "";
}