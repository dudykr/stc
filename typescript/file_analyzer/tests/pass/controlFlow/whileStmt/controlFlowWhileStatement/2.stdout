warning: Type
 --> $DIR/tests/pass/controlFlow/whileStmt/controlFlowWhileStatement/2.ts:2:1
  |
2 | x = "";
  | ^^^^^^
  |
  = note: ""

warning: Type
 --> $DIR/tests/pass/controlFlow/whileStmt/controlFlowWhileStatement/2.ts:3:12
  |
3 | while (x = x.length) {
  |            ^
  |
  = note: string

warning: Type
 --> $DIR/tests/pass/controlFlow/whileStmt/controlFlowWhileStatement/2.ts:3:12
  |
3 | while (x = x.length) {
  |            ^^^^^^^^
  |
  = note: number

warning: Type
 --> $DIR/tests/pass/controlFlow/whileStmt/controlFlowWhileStatement/2.ts:3:8
  |
3 | while (x = x.length) {
  |        ^^^^^^^^^^^^
  |
  = note: number

warning: Type
 --> $DIR/tests/pass/controlFlow/whileStmt/controlFlowWhileStatement/2.ts:4:5
  |
4 |     x; // number
  |     ^
  |
  = note: number

warning: Type
 --> $DIR/tests/pass/controlFlow/whileStmt/controlFlowWhileStatement/2.ts:5:5
  |
5 |     x = "";
  |     ^^^^^^
  |
  = note: ""

